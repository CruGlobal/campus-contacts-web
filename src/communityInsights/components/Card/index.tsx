import styled from '@emotion/styled';
import React from 'react';
import { withTheme } from 'emotion-theming';

const Container = styled.div`
    box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.2);
    background: ${({ theme }) => theme.colors.white};
    border-radius: 8px;
    padding-left: 24px;
    padding-right: 24px;
    padding-top: 43px;
    margin-top: 36px;
`;

const Title = styled.h1`
    font-weight: 300;
    font-size: 36px;
    line-height: 38px;
    color: ${({ theme }) => theme.colors.primary};
    margin-bottom: 12px;
    margin-top: 0px;
`;

interface SubtitleProps {
    noMarginBottom?: boolean;
}

const Subtitle = styled.h2<SubtitleProps>`
    font-size: 14px;
    line-height: 20px;
    font-weight: normal;
    color: ${({ theme }) => theme.colors.primary};
    margin-top: 0;
    margin-bottom: ${(props: SubtitleProps) =>
        props.noMarginBottom ? '-40px' : '0'};
`;

interface ContentProps {
    noPadding?: boolean;
}

const Content = styled.div<ContentProps>`
    padding: ${(props: ContentProps) => (props.noPadding ? 0 : '25px')};
    margin: ${(props: ContentProps) => (props.noPadding ? '0 -24px' : '0')};
`;

interface Props {
    title?: string;
    subtitle?: string;
    children?: any;
    noPadding?: boolean;
    noMarginBottom?: boolean;
}

const Card = (props: Props) => {
    const { title, subtitle, children, noPadding, noMarginBottom } = props;
    return (
        <Container>
            <Title>{title}</Title>
            <Subtitle noMarginBottom={noMarginBottom}>{subtitle}</Subtitle>
            <Content noPadding={noPadding}>{children}</Content>
        </Container>
    );
};

export default withTheme(Card);
