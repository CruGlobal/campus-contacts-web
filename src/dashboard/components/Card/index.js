import styled from '@emotion/styled';
import React from 'react';
import PropTypes from 'prop-types';

const Container = styled.div`
    box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.2);
    overflow: hidden;
    background: ${({ theme }) => theme.colors.white};
    border-radius: 8px;
    padding-left: 24px;
    padding-right: 24px;
    padding-top: 43px;
    margin-top: 36px;
`;

const Title = styled.h1`
    font-weight: 300;
    font-size: 36px;
    line-height: 38px;
    color: ${({ theme }) => theme.colors.primary};
    margin-bottom: 12px;
    margin-top: 0px;
`;

const Subtitle = styled.h2`
    font-size: 14px;
    line-height: 20px;
    font-weight: normal;
    color: ${({ theme }) => theme.colors.primary};
    margin-top: 0;
`;

const Content = styled.div`
    padding: 25px;
`;

const Card = props => {
    const { title, subtitle, children } = props;

    return (
        <Container>
            <Title>{title}</Title>
            <Subtitle>{subtitle}</Subtitle>
            <Content>{children}</Content>
        </Container>
    );
};

export default Card;

Card.propTypes = {
    title: PropTypes.string,
    subtitle: PropTypes.string,
    children: PropTypes.element,
};
