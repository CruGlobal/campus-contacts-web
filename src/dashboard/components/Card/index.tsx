import styled from '@emotion/styled';
import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'emotion-theming';

const Container = styled.div`
    box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.2);
    background: ${({ theme }: { theme: any }) => theme.colors.white};
    border-radius: 8px;
    padding-left: 24px;
    padding-right: 24px;
    padding-top: 43px;
    margin-top: 36px;
`;

const Title = styled.h1`
    font-weight: 300;
    font-size: 36px;
    line-height: 38px;
    color: ${({ theme }: { theme: any }) => theme.colors.primary};
    margin-bottom: 12px;
    margin-top: 0px;
`;

interface SubtitleProps {
    noMarginBottom: number;
}

const Subtitle = styled.h2<SubtitleProps>`
    font-size: 14px;
    line-height: 20px;
    font-weight: normal;
    color: ${({ theme }: { theme: any }) => theme.colors.primary};
    margin-top: 0;
    margin-bottom: ${(props: any) => (props.noMarginBottom ? '-40px' : '0')};
`;

interface ContentProps {
    noPadding: number;
}

const Content = styled.div<ContentProps>`
    padding: ${(props: any) => (props.noPadding ? 0 : '25px')};
    margin: ${(props: any) => (props.noPadding ? '0 -24px' : '0')};
`;

const Card = (props: any) => {
    const { title, subtitle, children, noPadding, noMarginBottom } = props;
    return (
        <Container>
            <Title>{title}</Title>
            <Subtitle noMarginBottom={noMarginBottom}>{subtitle}</Subtitle>
            <Content noPadding={noPadding}>{children}</Content>
        </Container>
    );
};

export default withTheme(Card);

Card.propTypes = {
    title: PropTypes.string,
    subtitle: PropTypes.string,
    children: PropTypes.element,
    noPadding: PropTypes.bool,
    noMarginBottom: PropTypes.bool,
};
