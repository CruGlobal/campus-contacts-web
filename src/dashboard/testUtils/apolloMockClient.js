import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
