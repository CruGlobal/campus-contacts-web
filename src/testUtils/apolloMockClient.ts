import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
