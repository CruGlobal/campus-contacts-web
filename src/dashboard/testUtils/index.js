import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
