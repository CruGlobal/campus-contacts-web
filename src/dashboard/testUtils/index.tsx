import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
