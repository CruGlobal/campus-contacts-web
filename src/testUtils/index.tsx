import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
