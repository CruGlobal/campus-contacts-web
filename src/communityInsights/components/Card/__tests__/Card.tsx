import React from 'react';

import { renderWithContext } from '../../../../testUtils';

import Card from '..';

describe('<Card />', () => {
    it('should render properly', async () => {
        renderWithContext(<Card />).snapshot();
    });

    it('should render properly with children', async () => {
        const child = <div>Children</div>;
        renderWithContext(<Card>{child}</Card>).snapshot();
    });

    it('should render properly with title and subtitle', async () => {
        renderWithContext(
            <Card title={'Title'} subtitle={'Subtitle'} />,
        ).snapshot();
    });
});
