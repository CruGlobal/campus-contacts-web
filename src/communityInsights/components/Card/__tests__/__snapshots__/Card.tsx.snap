// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render properly 1`] = `
.emotion-3 {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.2);
  background: #ffffff;
  border-radius: 8px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 43px;
  margin-top: 36px;
}

.emotion-0 {
  font-weight: 300;
  font-size: 36px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 12px;
  margin-top: 0px;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
  color: #505256;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-2 {
  padding: 25px;
  margin: 0;
}

<div>
  <div
    class="emotion-3"
  >
    <h1
      class="emotion-0"
    />
    <h2
      class="emotion-1"
    />
    <div
      class="emotion-2"
    />
  </div>
</div>
`;

exports[`<Card /> should render properly with children 1`] = `
.emotion-3 {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.2);
  background: #ffffff;
  border-radius: 8px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 43px;
  margin-top: 36px;
}

.emotion-0 {
  font-weight: 300;
  font-size: 36px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 12px;
  margin-top: 0px;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
  color: #505256;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-2 {
  padding: 25px;
  margin: 0;
}

<div>
  <div
    class="emotion-3"
  >
    <h1
      class="emotion-0"
    />
    <h2
      class="emotion-1"
    />
    <div
      class="emotion-2"
    >
      <div>
        Children
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> should render properly with title and subtitle 1`] = `
.emotion-3 {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.2);
  background: #ffffff;
  border-radius: 8px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 43px;
  margin-top: 36px;
}

.emotion-0 {
  font-weight: 300;
  font-size: 36px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 12px;
  margin-top: 0px;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
  color: #505256;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-2 {
  padding: 25px;
  margin: 0;
}

<div>
  <div
    class="emotion-3"
  >
    <h1
      class="emotion-0"
    >
      Title
    </h1>
    <h2
      class="emotion-1"
    >
      Subtitle
    </h2>
    <div
      class="emotion-2"
    />
  </div>
</div>
`;
